### PR TITLE
1610 menu defaults

### LINKED ISSUE #1610: [UX] 'Auto Menu Settings' in core and add Pages to the main menu by default
### Weekly update (19th Dec)

https://github.com/backdrop/backdrop/pull/2999 is RTBC, but more feedback/wording changes requested...

----

Drupal has long had this user-nightmare of lost pages. You made one, now where did it go? Yes, the experienced user knows about the content orphanage where all content is listed, but to a new user losing these pages can be traumatic. Backdrop now has this problem too.

If we placed basic pages into the primary navigation menu by default, then first-time users would be able to find their recently created content without needing to visit the content orphanage.

Experienced users could disable this behavior, or simply move pages on a case-by-case basis out of this menu.

## Proposed solution

* On the Page node form, check the box for "create menu item" by default.
* Add 2 per-content type settings: 1) create menu item by default, and 2) which menu.

---

Advocate for this issue: @BWPanda 

----

Original PR by @BWPanda: ~https://github.com/backdrop/backdrop/pull/2903~
Alternate PR by @jenlampton: ~https://github.com/backdrop/backdrop/pull/2951~
New PR by @BWPanda: https://github.com/backdrop/backdrop/pull/2999

### PR DESCRIPTION
**Pull request by jenlampton**
Fixes https://github.com/backdrop/backdrop-issues/issues/1610
Replaces https://github.com/backdrop/backdrop/pull/2951

First commit: Feature removal.
* Removes the option to set a Default `Menu link title`

Second commit: UX Updates.
* Changes checkbox label `Provide a default menu link` to `Add a link into the menu for new content`
* Removes description text from `Add a link into the menu by default` checkbox
* Changes checkboxes label `Available menus` to `Menus where this content may be placed`
* Removes description text for `Menus available for placing links`
* Updated description text for `Default parent item` to `Links to content will be placed below this item in the menu.`
* Changes the link `Description` field from textarea to textfield.
* adds `Often read by screen readers.` to link description `#description`.

### CODE REVIEW

File: core/modules/menu/menu.module [hunk 1]
```diff
@@ -744,6 +744,11 @@ function menu_node_submit(Node $node, $form, $form_state) {
   if (!empty($form_state['values']['menu']['parent'])) {
     list($node->menu['menu_name'], $node->menu['plid']) = explode(':', $form_state['values']['menu']['parent']);
   }
+
+  // Replace tokens in the menu link title.
```
Comment (BWPanda) [inline core/modules/menu/menu.module]: It seems we don't need this anymore, since that feature was removed...

File: core/modules/menu/menu.module [hunk 2]
```diff
@@ -744,6 +744,11 @@ function menu_node_submit(Node $node, $form, $form_state) {
...
+  if (!empty($form_state['values']['menu']['link_title'])) {
+    $node->menu['link_title'] = token_replace($form_state['values']['menu']['link_title'], array('node' => $node));
+  }
```
Comment (ghost) [inline core/modules/menu/menu.module]: It seems we don't need this anymore, since that feature was removed...

**Review by ghost — CHANGES_REQUESTED** [review]

File: core/modules/menu/menu.module (hunk 2 above)
Comment (jenlampton) [reply core/modules/menu/menu.module]: thank you!

**Comment by serundeputy** [comment]
closing to restart sandboxes ...

**Comment by serundeputy** [comment]
re-opening.

File: core/modules/menu/menu.module [hunk 3]
```diff
@@ -697,15 +697,16 @@ function menu_form_node_form_alter(&$form, $form_state) {
...
   $form['menu']['link']['description'] = array(
-    '#type' => 'textarea',
-    '#title' => t('Description'),
+    '#type' => 'textfield',
+    '#size' => 120,
+    '#title' => t('Description or clarification'),
     '#default_value' => isset($link['options']['attributes']['title']) ? $link['options']['attributes']['title'] : '',
     '#rows' => 1,
-    '#description' => t('Shown when hovering over the menu link.'),
+    '#description' => t('Shown when hovering over the menu link.') . ' ' . t('Often read by screen readers.'),
```
Comment (ghost) [inline core/modules/menu/menu.module:709]: Are these changes necessary for this issue? Sounds like this should be a separate issue, as I imagine there'd be other places that could use these changes too...
Also, I don't think the description text should be split between two `t()` functions. Seems unnecessary and harder to read (code-wise).

File: core/modules/views/plugins/views_plugin_display_page.inc [hunk 4]
```diff
@@ -264,7 +264,7 @@ class views_plugin_display_page extends views_plugin_display {
     $options['path'] = array(
       'category' => 'page',
       'title' => t('Path'),
-      'value' => views_ui_truncate($path, 24),
+      'value' => $path,
```
Comment (ghost) [inline core/modules/views/plugins/views_plugin_display_page.inc:267]: This is being discussed/addressed separately in https://github.com/backdrop/backdrop-issues/issues/1309

**Review by ghost — CHANGES_REQUESTED** [review]
I think there are some unnecessary changes here that should be left to separate issues...

**Comment by ghost** [comment]
I've created a new PR based off this one but without the code I mentioned above: https://github.com/backdrop/backdrop/pull/2999

**Comment by jenlampton** [comment]
Replaced by https://github.com/backdrop/backdrop/pull/2999
